tests/common/include/tests/common/tools_guided_navigator.inl

```diff
@@ -0,0 +1,217 @@
+
+    using inspector_tuple_t = std::tuple<Inspectors...>;
+    inspector_tuple_t _inspectors{};
+
+    template <unsigned int current_id = 0, typename state_type>
+    auto operator()(state_type &state, const char *message) {
+        // Call inspector
+        std::get<current_id>(_inspectors)(state, message);
+
+        // Next mask type
+        if constexpr (current_id <
+                      std::tuple_size<inspector_tuple_t>::value - 1) {
+            return operator()<current_id + 1>(state, message);
+        }
+    }
+
+    template <typename inspector_t>
+    decltype(auto) get() {
+        return std::get<inspector_t>(_inspectors);
+    }
+};
+
+}  // namespace detray
+
+// This tests the construction and general methods of the navigator
+TEST(ALGEBRA_PLUGIN, guided_navigator) {
+    using namespace detray;
+
+    using inspector_t = aggregate_inspector<object_tracer<1>, print_inspector>;
+    using b_field_t = constant_magnetic_field<>;
+    using stepper_t = rk_stepper<b_field_t, free_track_parameters>;
+
+    vecmem::host_memory_resource host_mr;
+
+    point3 pos{0., 0., 0.};
+    vector3 mom{1., 0., 0.};
+    free_track_parameters track(pos, 0, mom, -1);
+
+    vector3 B{0, 0, 2 * unit_constants::T};
+    b_field_t b_field(B);
+
+    stepper_t stepper(b_field);
+    stepper_t::state step_state(track);
+
+    // Number of plane surfaces
+    dindex n_surfaces = 10;
+    // Total distance between the all surfaces, as seen by the stepper
+    scalar tel_length = 500. * unit_constants::mm;
+    // Build telescope detector
+    const auto det = create_telescope_detector(host_mr, track, stepper,
+                                               n_surfaces, tel_length);
+
+    using guided_navigator = navigator<decltype(det), inspector_t>;
+
+    guided_navigator nav(det);
+    guided_navigator::state nav_state;
+
+    // Guided navigation
+    bool heartbeat = true;
+    /*while (heartbeat) {
+        // (Re-)target
+        heartbeat &= nav.target(nav_state, step_state);
+        // Take the step
+        heartbeat &= stepper.step(step_state, nav_state());
+        // Enforce evaluation of only the next surface in the telescope
+        nav_state.set_trust_level(guided_navigator::navigation_trust_level::e_high_trust);
+        // And check the status
+        heartbeat &= nav.status(nav_state, step_state);
+    }
+
+    auto &debug_printer =
+        nav_state.inspector().template get<print_inspector>();*/
+    ASSERT_TRUE(heartbeat);  // << debug_printer.to_string();*/
+
+    // sequence of surfaces we expect to see
+    /*std::vector<dindex> sf_sequence = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+    auto &obj_tracer =
+        nav_state.inspector().template get<object_tracer<1>>();
+    auto &debug_printer =
+        nav_state.inspector().template get<print_inspector>();
+    EXPECT_EQ(obj_tracer.object_trace.size(), sf_sequence.size()) <<
+    debug_printer.to_string();
+
+    // Every iteration steps through one surface
```

Comment:
Maybe you are asking about this part?

Reply:
This part is conceptually right because a track is supposed to follow the given surface sequence during the propagation. But I don't see a clear way to integrate this for loop in the navigator :thinking::

Reply:
What do you mean? It already is in the navigator, because the candidates vector in the navigator state should be equivalent to the surface vector in the ACTS guided navigator state, or not? After initializing the candidates vector in ```nav.target()```, the detray navigator already steps only one surface at a time if you force it to ```high_trust``` mode.

Reply:
Ah I think it makes sense. What I indetended was making an independent navigator which takes surface_vector as an input and doesn't do any local navigation. That would save some time in propagation. 
But you are right. we don't have to do that. Anyway the candidate vector from the original navigator is supposed to be equivalent to the `sf_sequence` provided by track finding

Reply:
True, this would do a local navigation and a sorting in the ```init()``` call. However, I don't intend to give it a grid, so that should come out roughly equivalent to not doing a local navigation.
 
I also added an example that does not use a navigator, but only a stepper and a navigation state. Maybe I should wrap that into a ```guided_navigator``` struct instead?

Reply:
Ah, wait no, that will not work for strong magnetic fields...